berth_reservations/logging.py

```diff
@@ -0,0 +1,49 @@
+import logging
+import time
+
+logger = logging.getLogger("requests")
+
+
+__all__ = ["LogRequestFilter", "RequestLogger"]
+
+
+class LogRequestFilter(logging.Filter):
+    """Since the requests are being logged by the RequestLogger,
+    we filter the django.server logs to avoid having duplicate logs"""
+
+    def filter(self, record: logging.LogRecord):
+        return record.name != "django.server"
+
+
+class RequestLogger:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        exec_time = time.time()
+        response = self.get_response(request)
+        exec_time = int((time.time() - exec_time) * 1000)
+
+        path = request.get_full_path()
+        status = response.status_code
+        method = request.method
+
+        message = f"{method} {path} {status}"
+        context = {
+            "host": request.get_host(),
+            "method": method,
+            "agent": request.headers["USER_AGENT"],
+            "body": request.body.decode(request.encoding or "utf-8"),
```

Comment:
Is it possible for `request.body` to be `None`? For example `GET` requests?

Reply:
It might be safer to wrap everything in a try/catch here. If this middleware raises for any reason the client will never get a response back?

Reply:
It will log an empty string.

Reply:
Most of the getters should be safe, but I wrapped in a `try/except` to be on the safe side.